ProcessMaker/processmaker
# FOUR-5882:   Terminate event not closing all open tasks

## Issue & Reproduction Steps

-     Import the process
-     Run a new request, you should see 3 tasks: Task1, Task2 and Terminate task
-     Open "Terminate task" and submit it

Even when the Terminate End Event is triggered, Task 1 and Task 2 are still active. A terminate should close all open tokens of a request.

## Solution
- Added listener that persists closed tokens during a terminate end event.

## How to Test
Follow the steps to reproduce and:
- Verify that when submitting "Task 1" it arrives to a normal end event and  just Task 1 is closed. Task 2 and "Terminate Task" should be active
- Verify that when submitting "Terminate Task", all  tasks are closed and the request completed

## Related Tickets & Packages
- [https://processmaker.atlassian.net/browse/FOUR-5882](https://processmaker.atlassian.net/browse/FOUR-5882)

## Code Review Checklist
